lgc/include/lgc/state/PipelineState.h

```diff
@@ -446,6 +454,7 @@ class PipelineState final : public Pipeline {
   std::unique_ptr<InterfaceData> m_interfaceData[ShaderStageCompute + 1] = {}; // Per-shader InterfaceData
   PalMetadata *m_palMetadata = nullptr;                                        // PAL metadata object
   unsigned m_waveSize[ShaderStageCountInternal] = {};                          // Per-shader wave size
+  bool m_inOutPackStates[ShaderStageGfxCount][2] = {}; // The input and output packable state of each shader stage
```

Comment:
`2 -> static_cast<unsigned>(PackType::Count)`

Reply:
I think it is unnecessarily complicated having a single array here and the enum to access the second dimension. Instead we should have two arrays m_inputPackStates and m_outputPackStates, and two separate functions to access them. If you follow my suggestion to unroll that little loop in PatchResourceCollect, then only one place remains that has a variable LocationType index, in an assert, and that can easily be changed to a conditional expression.

Reply:
Follow Tim's suggestions, two arrays are used to represent the state of input and output packable locations, respectively. Done.